tests/integration/TestKafkaProducer.cc

```diff
@@ -131,8 +133,8 @@ TEST(KafkaSyncProducer, FailToSendMessagesWithAcksAll)
     {
         auto record = ProducerRecord(topic, Key(msg.first.c_str(), msg.first.size()), Value(msg.second.c_str(), msg.second.size()));
         std::cout << "[" << Utility::getCurrentTime() << "] ProducerRecord: " << record.toString() << std::endl;
-        // "no in-sync replica" for the topic.
-        EXPECT_KAFKA_THROW(producer.send(record), RD_KAFKA_RESP_ERR_NOT_ENOUGH_REPLICAS);
+        // Since "no in-sync replica" for the topic, it would keep trying
+        EXPECT_KAFKA_THROW(producer.send(record), RD_KAFKA_RESP_ERR__MSG_TIMED_OUT);
```

Comment:
@edenhill It seems the behavior changed in librdkafka `v1.5.2`? -- if the topic is with `replication-factor=1`, and the cluster is configured with `min.insync.replicas=1`, the error would be `MSG_TIMED_OUT`, instead of `NOT_ENOUGH_REPLICAS`?

Reply:
NOT_ENOUGH_REPLICAS is a retriable error so the producer will retry producing the message after a backoff until the number of retries are exhausted.
The default number of `retries` changed from 2 to infinity in v1.5.2, which means the message will time out rather than fail due to retries, that's why you are seeing MSG_TIMED_OUT.

Reply:
Thank you, @edenhill, I see, it makes sense. 👍